processing/src/main/java/io/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -107,6 +107,15 @@ public TimeseriesQueryQueryToolChest(
           Map<String, Object> context
       )
       {
+        int limit = ((TimeseriesQuery) queryPlus.getQuery()).getLimit();
+        if (limit < Integer.MAX_VALUE) {
+          return super.doRun(
```

Comment:
It looks like the two calls to `super.doRun` are identical, and nontrivial, so please just call it one time. It will make the code easier to read and maintain later.

You could do a style like you're doing in TimeseriesQueryEngine: get the sequence first and then return it with or without a limit.